app/services/exports/export_distributions_csv_service.rb

```diff
@@ -114,15 +115,7 @@ def base_headers
     def item_headers
       return @item_headers if @item_headers
 
-      item_names = Set.new
-
-      distributions.each do |distribution|
-        distribution.line_items.each do |line_item|
-          item_names.add(line_item.item.name)
-        end
-      end
-
-      @item_headers = item_names.sort
+      @item_headers = @organization.items.uniq.sort_by(&:created_at).pluck(:name)
```

Comment:
This isn't doing what you think it is :) It's first loading all items into memory and then sorting and mapping them. I think you wanted

```ruby
@item_headers = @organization.items.order(:created_at).distinct.select([:created_at, :name]).map(&:name)
```

which does it purely in the database.

Reply:
For posterity: My comment came across as condescending. A better wording would be something like: "This is doing queries in the app instead of in the database; it would be much faster to do the query this way."